docker-compose-dev.yml

```diff
@@ -0,0 +1,29 @@
+services:
+  db:
+    image: postgres:16
+    expose:
+      - 5432
+    environment:
+      PGDATA: /var/lib/postgresql/data/pgdata
+      # Defaults for createdb/dropdb/pg_restore etc.
+      PGDATABASE: ons
+      PGUSER: ons
+      # Values for initdb
+      POSTGRES_HOST_AUTH_METHOD: trust
+      POSTGRES_USER: ons
+    volumes:
+      - pgdata:/var/lib/postgresql/data
+    attach: false
+    ports:
+      - 5432:5432
+
+  redis:
+    image: redis:7.2
+    expose:
+      - 6379
+    attach: false
+    ports:
+      - 6379:6379
```

Comment:
If you change the env vars to use localhost, you don't need both `expose` and `ports`. It can be misleading to define both.

Reply:
I think that the right way to do here is to remove the `expose` bit and leave the `ports`.

If I understand it correctly the `expose` is analogous to the [EXPOSE in a Dockerfile](https://docs.docker.com/reference/dockerfile/#expose) in that:
> "The EXPOSE instruction doesn't actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published." 

and that the actual port forwarding needs to be set up as a setting when running `docker run` or in the `ports` bit in a compose file.

An experiment confirmed that in fact the `expose` is not necessary so I'm going to remove it.